src/modules/Resources/devices.types.ts

```diff
@@ -91,52 +91,65 @@ interface DeviceCreateInfoLegacy extends Omit<DeviceCreateInfoBasic, "type"> {
 type DeviceCreateInfo = DeviceCreateInfoLegacy | DeviceCreateInfoMutable | DeviceCreateInfoImmutable;
 type DeviceEditInfo = Partial<Omit<DeviceCreateInfo, "chunk_period" | "type"> & { chunk_retention: number }>;
 
-interface DeviceInfo extends Required<Omit<DeviceCreateInfoBasic, "configuration_params">> {
+interface DeviceInfoBase {
   /**
    * Device ID.
    */
   id: GenericID;
-  /**
-   * Device's data storage (bucket) type.
-   */
-  type: DataStorageType;
   /**
    * ID of the profile that owns the device.
    */
   profile: GenericID;
   /**
    * Bucket storing the device's data.
+   *
+   * @deprecated
    */
   bucket: {
     id: GenericID;
     name: string;
   };
   /**
-   * Date for the device's last output.
+   * Payload parser.
+   *
+   * Encoded string when enabled, `null` when not enabled.
    */
-  last_output: Date | null;
+  payload_decoder: string | null;
   /**
-   * Date for the device's last input.
+   * Date for the device's creation.
    */
-  last_input: Date | null;
+  created_at: Date;
   /**
    * Date for the device's last update.
    */
   updated_at: Date;
   /**
-   * Date for the device's creation.
-   */
-  created_at: Date;
-  /**
-   * Date for the device's last inspection.
+   * Date for the device's last input.
    */
-  inspected_at: Date | null;
+  last_input: Date | null;
   /**
-   * Date for the device's last data retention.
+   * Device-specific soft limits on RPM.
+   *
+   * `rpm` is `null` when not set or when the profile doesn't have the add-on.
    */
-  last_retention: Date | null;
+  rpm: {
+    data_input?: number;
+    data_output?: number;
+  } | null;
 }
 
+type DeviceInfoImmutable = Required<
+  DeviceInfoBase &
+    Omit<DeviceCreateInfoImmutable, "configuration_params" | "serie_number" | "connector_parse" | "parse_function">
```

Comment:
I don't remember some of these parameters, so I left them on `Omit` since the API did not return any of those for me.